src/core/ext/filters/client_channel/lb_policy.h

```diff
@@ -279,7 +279,7 @@ class LoadBalancingPolicy : public InternallyRefCounted<LoadBalancingPolicy> {
 
     /// Creates a new subchannel with the specified channel args.
     virtual RefCountedPtr<SubchannelInterface> CreateSubchannel(
-        ServerAddress address, const grpc_channel_args& args) = 0;
+        ServerAddress address, const ChannelArgs& args) = 0;
```

Comment:
The decision to use `const&` everywhere was carefully made with measurements of bloat.

`std::move` would mean that we need to do some work at every call site (to null the sending object and create the receiving object) and oftentimes also call a destructor - whereas `const&` makes the passing of channel args free - since we're just moving a pointer.

With the way that we've plumbed channel args through many small forwarding functions this makes a relatively large difference in overall bloat.

Reply:
Interesting!  I didn't realize that would add so much bloat, but it makes sense.